debezium-embedded/src/main/java/io/debezium/embedded/async/ParallelSmtAndConvertAsyncConsumerProcessor.java

```diff
@@ -40,14 +39,20 @@ public class ParallelSmtAndConvertAsyncConsumerProcessor<R> extends AbstractReco
     @Override
     public void processRecords(final List<SourceRecord> records) throws Exception {
         LOGGER.debug("Thread {} is submitting {} records for processing.", Thread.currentThread().getName(), records.size());
-        final List<Future<Void>> recordFutures = new ArrayList<>(records.size());
-        records.stream().forEachOrdered(
-                r -> recordFutures.add(recordService.submit(new ProcessingCallables.TransformConvertConsumeRecord<>(r, transformations, convertor, consumer))));
+
+        final Future<Void>[] recordFutures = new Future[records.size()];
+        int i = 0;
```

Comment:
Given that the `i` is no longer local to `for` would it be possible to give it a more descriptive name here  and in other parts of the code too?

Reply:
sure, changed to `arrayIndex`